search-commons/src/main/java/no/unit/nva/search/ticket/TicketAccessFilter.java

```diff
@@ -0,0 +1,276 @@
+    }
+
+    /**
+     * Exclude these ticket types.
+     *
+     * @param ticketTypes ticket types to exclude
+     * @return TicketQuery (builder pattern)
+     * @apiNote ONLY SET THIS MANUALLY IN TESTS
+     */
+    public TicketAccessFilter excludeTypes(TicketType... ticketTypes) {
+        return excludeTypes(listToEnumSet(ticketTypes));
+    }
+
+    /**
+     * Filter on organization.
+     *
+     * @param organization organization id
+     * @return TicketQuery (builder pattern)
+     * @apiNote ONLY SET THIS MANUALLY IN TESTS
+     */
+    public TicketAccessFilter organization(URI organization) {
+        this.organizationId = organization;
+        return this;
+    }
+
+    public String getCurrentUser() {
+        return currentUser;
+    }
+
+    protected DisMaxQueryBuilder getTicketTypeUserNameFilter() {
+        return QueryBuilders.disMaxQuery()
+                .add(new TermQueryBuilder(OWNER_USERNAME, currentUser))
+                .add(new TermsQueryBuilder(TYPE_KEYWORD, allowedTicketTypes))
+                .queryName(ANY_OF_TICKET_TYPE_OR_USER_NAME);
+    }
+
+    protected QueryBuilder getOrgFilter() {
+        return new AcrossFieldsQuery<TicketParameter>()
+                .buildQuery(ORGANIZATION_ID, organizationId.toString())
+                .findFirst()
+                .orElseThrow()
+                .getValue()
+                .queryName(ORGANIZATION_ID.asCamelCase());
+    }
+
+    protected BoolQueryBuilder getExcludedFilter() {
+        var excludedSet =
+                nonNull(excludeTicketTypes)
+                        ? excludeTicketTypes
+                        : Arrays.stream(TicketType.values())
+                                .filter(f -> !allowedTicketTypes.contains(f))
+                                .collect(
+                                        Collectors.toCollection(
+                                                () -> EnumSet.noneOf(TicketType.class)));
+        return boolQuery()
+                .mustNot(termsQuery(TYPE_KEYWORD, excludedSet))
+                .queryName(NOT_ANY_OF_TICKET_TYPE);
+    }
+
+    private TicketAccessFilter includedTypes(Set<TicketType> ticketTypes) {
+        this.allowedTicketTypes = ticketTypes;
+        return this;
+    }
+
+    private TicketAccessFilter excludeTypes(Set<TicketType> ticketTypes) {
+        this.excludeTicketTypes = ticketTypes;
+        return this;
+    }
+
+    /**
+     * Apply business rules to determine which ticket types are allowed.
+     *
+     * <ul>
+     *   <lh>Business Rules.</lh>
+     *   <li>manage_doi -> DOI_REQUEST
+     *   <li>support -> GENERAL_SUPPORT_CASE
+     *   <li>manage_publishing_requests -> PUBLISHING_REQUEST
+     *   <li>manage_customers -> DOI_REQUEST, GENERAL_SUPPORT_CASE, PUBLISHING_REQUEST
+     *   <li>is_owner -> DOI_REQUEST, GENERAL_SUPPORT_CASE, PUBLISHING_REQUEST
+     * </ul>
+     */
+    private Set<TicketType> applyBusinessRules(List<AccessRight> accessRights)
```

Comment:
- Method name should describe what method does -> it returns tickets types user has right to search based on access rights. Rename it to something simple as getAllowedTicketTypes(accessRights).

- Should this filter throw exception? Or return empty ticket list when no access rights to search tickets with "not allowed" params?

Reply:
The reason I renamed it, was that access right validation was handled here and there in this class.
Now it should be in only one place....

Reply:
I feel like the response here is tangential to what was being asked — am I missing something?